### PR TITLE
adding lm and cv_cls task scripts

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
1. Adding lm and cv_cls task scripts
2. CV Classification task is slower and has lower eval performance when using `inductor` backend (a single run with the launch command given in README.md)
Screenshot 2022-11-22 at 7 06 57 PM
3. For causal language modeling, everything looks as expected, i.e., `inductor` backend preserves performance and has good speedup (3X).
Screenshot 2022-11-22 at 7 31 32 PM

### CODE REVIEW

File: scripts/cv_classification.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        datasets.utils.logging.set_verbosity_error()
+        transformers.utils.logging.set_verbosity_error()
+
+    dataset = load_dataset(args.dataset_name, task="image-classification")
+
+    # Prepare label mappings.
+    # We'll include these in the model's config to get human readable labels in the Inference API.
+    labels = dataset["train"].features["labels"].names
+    label2id = {label: str(i) for i, label in enumerate(labels)}
+    id2label = {str(i): label for i, label in enumerate(labels)}
```
Comment (sgugger) [inline scripts/cv_classification.py]: We don't care about the labels.

File: scripts/cv_classification.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    #
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    config = AutoConfig.from_pretrained(
+        args.model_name_or_path,
+        num_labels=len(labels),
+        i2label=id2label,
+        label2id=label2id,
+        finetuning_task="image-classification",
+    )
```
Comment (sgugger) [inline scripts/cv_classification.py]: No need to instantiate the config separately

File: scripts/cv_classification.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    feature_extractor = AutoFeatureExtractor.from_pretrained(args.model_name_or_path)
+    model = AutoModelForImageClassification.from_pretrained(
+        args.model_name_or_path,
+        from_tf=bool(".ckpt" in args.model_name_or_path),
+        config=config,
```
Comment (sgugger) [inline scripts/cv_classification.py]: Remove and put `num_labels` here.

File: scripts/cv_classification.py (hunk 1 above)
Comment (pacman100) [reply scripts/cv_classification.py]: Done

File: scripts/cv_classification.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        transformers.utils.logging.set_verbosity_info()
+    else:
+        datasets.utils.logging.set_verbosity_error()
+        transformers.utils.logging.set_verbosity_error()
+
+    dataset = load_dataset(args.dataset_name, task="image-classification")
+    feature_extractor = AutoFeatureExtractor.from_pretrained(args.model_name_or_path)
+    model = AutoModelForImageClassification.from_pretrained(
+        args.model_name_or_path,
+        from_tf=bool(".ckpt" in args.model_name_or_path),
```
Comment (sgugger) [inline scripts/cv_classification.py]:
```suggestion
```

File: scripts/cv_classification.py [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+        num_labels=len(dataset["train"].features["labels"].names),
+        finetuning_task="image-classification",
```
Comment (sgugger) [inline scripts/cv_classification.py]:
```suggestion
```

File: scripts/cv_classification.py [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+        eval_dataset = dataset["validation"].with_transform(preprocess_val)
+
+    # DataLoaders creation:
+    def collate_fn(examples):
+        pixel_values = torch.stack([example["pixel_values"] for example in examples])
+        labels = torch.tensor([example["labels"] for example in examples])
+        return {"pixel_values": pixel_values, "labels": labels}
+
+    train_dataloader = DataLoader(train_dataset, shuffle=True, collate_fn=collate_fn, batch_size=args.batch_size)
+    eval_dataloader = DataLoader(eval_dataset, collate_fn=collate_fn, batch_size=args.batch_size)
```
Comment (sgugger) [inline scripts/cv_classification.py]: You need to drop the last batches here.

File: scripts/language_modeling.py [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        )
+
+    train_dataset = lm_datasets["train"]
+    eval_dataset = lm_datasets["validation"]
+
+    # DataLoaders creation:
+    train_dataloader = DataLoader(
+        train_dataset, shuffle=True, collate_fn=default_data_collator, batch_size=args.batch_size
+    )
+    eval_dataloader = DataLoader(eval_dataset, collate_fn=default_data_collator, batch_size=args.batch_size)
```
Comment (sgugger) [inline scripts/language_modeling.py]: Same here.

Comment (pacman100) [reply scripts/language_modeling.py]: Done

**Review by sgugger — APPROVED** [review]
Thanks a lot!
